main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -132,7 +129,7 @@ object Resolver extends Phase[NamedAst.Root, ResolvedAst.Root] {
   /**
     * Performs name resolution on the given signature `s0` in the given namespace `ns0`.
     */
-  def resolve(s0: NamedAst.Sig, ns0: Name.NName, root: NamedAst.Root, sigs: SigLookup)(implicit flix: Flix): Validation[ResolvedAst.Sig, ResolutionError] = s0 match {
+  def resolve(s0: NamedAst.Sig, ns0: Name.NName, root: NamedAst.Root, sigs: Map[Name.NName, Map[String, NamedAst.Sig]])(implicit flix: Flix): Validation[ResolvedAst.Sig, ResolutionError] = s0 match {
```

Comment:
Is there a reason why we cannot pull out the sigs from root? 

Do they change depending on some context?

Reply:
I have the separate map as a shortcut lookup, so that we don't have to scan everything. But this map can be put as a parameter in root. I didn't do that originally since it has some redundancies with the `classes` field.